feat(curriculum): Build a Zoo Animal Scatterplot (#66379)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-build-a-zoo-animal-scatterplot/69cfe8a2c376f8d915f96462.md

---
id: 69cfe8a2c376f8d915f96462
title: Step 8
challengeType: 0
dashedName: step-8
---

# --description--

> You should recall `drawScatterplot` function in each window resize, because you want to update your scatterplot with changing window size!  

Reviewer: Exclamation point not need at the end here. Also, this wording sounds smoother on the objective for this step.

```suggestion
In this step, you will update your scatterplot with the changing window size.
```

For this add a event listener on `window`, and trigger `drawScatterplot` function each time. 

> So, call the `drawScatterplot` for the first time and add event listener on resize of window that triggers `drawScatterplo` function.

Reviewer: ```suggestion
Start by calling the `drawScatterplot` function.

Then below that function call, add an event listener to the `window` object.  The first argument for the `addEventListener()` will be the `"resize"` event and the second argument will be a reference to the `drawScatterplot` function.
```

An example of addEventListener could be : 

```js
function drawScatterplot() {
  console.log("Scatterplot drawn!");
}

window.addEventListener("resize", drawScatterplot);

> ```

Reviewer: I would remove this since campers already know how to work with event listeners from the JS cert.

Author: Make sense, I'm restructuring it. Just keeping the instructions for campers of what to do.

# --hints--

> You Should call the `drawScatterplot` function.

Reviewer: ```suggestion
You should call the `drawScatterplot` function.
```

```js
assert.match(code, /drawScatterplot\(\s*\)\s*;/);
```

> You should add event listener on resize of window that triggers `drawScatterplot` function.

Reviewer: ```suggestion
Your first argument for the `addEventListener()` should be the `"resize"` event and the second argument should be a reference to the `drawScatterplot` function.
```

```js

> assert.match(code, /window\.addEventListener\(\s*"resize",\s*drawScatterplot\s*\)/);

Reviewer: the tests needs to accept both single and double quotes around resize.
